tests/conftest.py

```diff
@@ -257,5 +267,33 @@ def build_header(path, method, role='submitter'):
         request.headers = dict(request.headers)
         auth.__call__(request)
         return request.headers
+
     return build_header
 
+
+@pytest.fixture(scope='session')
+def dictionary_setup():
+    def build_dict():
```

Comment:
If passing the `build_dict` function in the result, then it could make sense to allow additional URLs to be patched by the caller. Otherwise though, if this will only need to patch the single URL for s3, this should work fine.

Reply:
Added param for build_dict.